src/base/QXmppJingleData.cpp

```diff
@@ -1078,27 +1084,36 @@ void QXmppJingleIq::Reason::parse(const QDomElement &element)
         if (child.namespaceURI() == ns_jingle_rtp_errors) {
             if (const auto index = JINGLE_RTP_ERROR_CONDITIONS.indexOf(child.tagName());
                 index != -1) {
-                m_rtpErrorCondition = RtpErrorCondition(index);
+                d->m_rtpErrorCondition = RtpErrorCondition(index);
             }
             break;
         }
     }
+
+    if (!element.namespaceURI().isEmpty() && !element.parentNode().isNull() && element.parentNode().namespaceURI() != element.namespaceURI()) {
+        d->m_namespaceUri = element.namespaceURI();
+    }
```

Comment:
* please remove the attribute and always serialize with `ns_jingle`
* one issue with the current approach is that no namespace is set when you create a new JingleReason and serialize that
* you can set the namespace manually with the namespaceUri setter of course, but you can forget that very easily and produce XML that other clients don't understand
* namespaceUri accepts any namespace, but the jingle reason element has a standardized xml namespace
* serializing `ns_jingle` although the parent has the namespace already set is no problem